### PR TITLE
feat: support org and user metadata

### LINKED ISSUE #29: Support for org metadata
Zitadel provides an api to manipulate metadata on org level. The provider should also provide this feature.

### LINKED ISSUE #132: Manage user metadata
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

I can manage user metadata through API but cannot do it through terraform
https://zitadel.com/docs/apis/resources/mgmt/management-service-set-user-metadata

### Describe your ideal solution

```
resource "zitadel_user_metadata" "default" {
  org_id            = data.zitadel_org.default.id
  user_id          =  data.zitadel_user.default.id
  key                = "example"
  value             = "value"
}
```

### Version

1.0.2

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by eliobischof**
This feature adds the Terraform resources *zitadel_org_metadata* and *zitadel_user_metadata*.
One resource instance manages exactly one metadata key-value pair in ZITADEL. This enables having some keys managed by Terraform and others outside of Terraform.
If the value for the key or the user_id property changes, the old key-value pair is deleted and a new key-value pair is created.

Closes #29 
Closes #132 

### Definition of Ready

- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] All non-functional requirements are met
- [x] The generic lifecycle acceptance test passes for affected resources.
- [x] Examples are up-to-date and meaningful. The provider version is incremented.
- [x] Docs are generated.
- [x] Code is generated where possible.

### CODE REVIEW

File: zitadel/org_metadata/funcs.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	if !ok {
+		return diag.Errorf("failed to get client")
+	}
+	client, err := helper.GetManagementClient(clientinfo)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	key := d.Get(KeyVar).(string)
+	value, err := helper.Base64Decode(d.Get(ValueVar).(string))
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/org_metadata/funcs.go:24]: Is that true? That would surprise me, as it takes a byte array, not a string.

I will verify that.

[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/org_metadata/funcs.go:24]: I compared what the console shows with and w/o decoding he value before sending it.

Without decoding:
image

With decoding:
image

image

Terraform doesn't support byte array field types, that's why I decided to use a base64 encoded value in the Terraform resource definition, which is a string.

[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/org_metadata/funcs.go:24]: That's exactly what I mean, why do you decode it? I only went with the documentation under https://zitadel.com/docs/apis/resources/mgmt/management-service-set-org-metadata with "The value has to be base64 encoded.". The only thing that you achieve with this is that you are forced to have a base64 encoded value in your terraform resource, and the value is then always decoded in ZITADEL, which could result in values that you have to encode and decode 2 times.

[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/org_metadata/funcs.go:24]: The grpc method takes a byte array directly and in the ZITADEL code I don't see that it is decoded from base64 anywhere. Only the http endpoint decodes a base64 string. The docs are maybe not perfectly clear about that.

**Review by stebenz — APPROVED** [review]

**Pull request merged**